app/controllers/report_controller.rb

```diff
@@ -354,6 +288,71 @@ def cancel_import
 
   private ###########################
 
+  def set_active_elements(feature)
+    if feature
+      self.x_active_tree ||= feature.tree_list_name
+      self.x_active_accord ||= feature.accord_name
+    end
+    get_node_info
+  end
+
+  def features
+    [{:role        => "miq_report_saved_reports",
+      :role_any    => true,
+      :name        => :savedreports,
+      :accord_name => "savedreports",
```

Comment:
@h-kataria : right, understood, the pattern went through some improvements already

but we should follow the simplest example and try to unify things

If you want to do those changes in a separate PR, i am not against.